scripts/install/install_clang_format

```diff
@@ -24,6 +24,13 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 set -o errexit
 
+if ! clang-format -version >/dev/null 2>&1; then
+  echo "Fixing permissions for brew install..."
+  sudo chown -R $(whoami) /usr/local/share /usr/local/bin
```

Comment:
This seems dangerous. :/
Can we change the install directory for brew executables?

Reply:
This is unfortunately the recommended solution from the brew community :(

https://stackoverflow.com/questions/16432071/how-to-fix-homebrew-permissions

Am open to alternative suggestions though for managing our software installs.

Reply:
I think we should just tell people how to install and let them manage permissions?

At the least:
* Don't do recursive
* Restore ownership to the original value
* Don't use $(whoami). I'm terrified that it could be an attack vector (by defining the shell variable `whoami` before invoking this).

Reply:
That's fair - am open to transforming our installers to just be a setup guide rather than an automated solution.